group_project_v2/utils.py

```diff
@@ -95,4 +98,91 @@ def mean(value_array):
         return None
 
 
+def outsider_disallowed_protected_view(func):
+    @functools.wraps(func)
+    def wrapper(*args, **kwargs):
+        try:
+            return func(*args, **kwargs)
+        except OutsiderDisallowedError as ode:
+            error_fragment = Fragment()
+            error_fragment.add_content(
+                loader.render_template('/templates/html/loading_error.html', {'error_message': unicode(ode)}))
+            error_fragment.add_javascript(loader.load_unicode('public/js/group_project_error.js'))
+            error_fragment.initialize_js('GroupProjectError')
+            return error_fragment
+
+    return wrapper
+
+
+def outsider_disallowed_protected_handler(func):
+    @functools.wraps(func)
+    def wrapper(*args, **kwargs):
+        try:
+            return func(*args, **kwargs)
+        except OutsiderDisallowedError as ode:
+            return {
+                'result': 'error',
+                'message': ode.message
+            }
+
+    return wrapper
+
+
+def key_error_protected_handler(func):
+    @functools.wraps(func)
+    def wrapper(*args, **kwargs):
+        try:
+            return func(*args, **kwargs)
+        except KeyError as exception:
+            log.exception("Missing required argument {}".format(exception.message))
+            return {'result': 'error', 'msg': ("Missing required argument {}".format(exception.message))}
+
+    return wrapper
+
+
+def log_and_suppress_exceptions(func):
+    @functools.wraps(func)
+    def wrapper(*args, **kwargs):
+        try:
+            return func(*args, **kwargs)
+        except Exception as exc:  # pylint: disable=broad-except
+            log.exception(exc)
+            return None
+
+    return wrapper
+
+
+def get_default_stage(stages):
+    """
+    Gets "default" stage from collection of stages. "Default" stage is displayed if not particular stage
+    was requested. Rules for getting "default" stage are as follows:
+        1. If all the stages are not open - sequentially first
+        2. If all the stages are closed - sequentially last
+        3. If at least one stage is open and not closed - sequentially last opened and not closed
+
+    """
+    if not stages:
+        return None
+
+    if all(not stage.is_open for stage in stages):
```

Comment:
@bradenmacdonald I thought I've fixed it - thank you for noticing that, I'll double check it.